evalml/pipelines/component_graph.py

```diff
@@ -366,7 +282,7 @@ def _compute_features(self, component_list, X, y=None, fit=False):
                     output = component_instance.predict(input_x)
                 else:
                     output = None
-                output_cache[component_name] = output
+                output_cache[f"{component_name}.x"] = output
```

Comment:
This change means that Estimators' outputs will be specified via `EstimatorName.x` rather than `EstimatorName`. I think this is clearer, since we are moving away from the expectation that just using a component name will default to the features / .x, now that we have more components which return valid .y values :)